src/announcements.ts

```diff
@@ -15,37 +15,44 @@ export class Announcer {
 		return document.getElementById(this.id);
 	}
 
-	createRegion(): Element {
+	createRegion(): HTMLElement {
 		const liveRegion = createElement(
 			`<p aria-live="assertive" aria-atomic="true" id="${this.id}" style="${this.style}"></p>`
 		);
 		document.body.appendChild(liveRegion);
 		return liveRegion;
 	}
 
-	announce(message: string, delay: number = 0) {
-		setTimeout(() => {
-			// // Fix screen readers not announcing the same message twice
-			if (this.region.textContent === message) {
-				message = `${message}.`;
-			}
-			// Clear before announcing
-			this.region.textContent = '';
-			this.region.textContent = message;
-		}, delay);
+	announce(message: string, delay: number = 0): Promise<void> {
+		return new Promise((resolve) => {
+			setTimeout(() => {
+				// // Fix screen readers not announcing the same message twice
```

Comment:
Maybe this code comment could be optimized (apart from the double `//`):

```js
// Fix screen readers not announcing the same message again
```

Reply:
True, updated it with a description of the actual intent.